flocker/common/test/test_interface.py

```diff
@@ -0,0 +1,158 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Tests for ``flocker.common.interface_decorator``.
+"""
+
+from twisted.trial.unittest import SynchronousTestCase
+
+from eliot.testing import (
+    LoggedMessage, assertContainsFields, capture_logging
+)
+from eliot import Field, MessageType
+
+from zope.interface import Interface, implementer
+
+from ..interface import interface_decorator
+
+
+# Eliot structures for testing ``interface_decorator``.
+METHOD = Field.for_types(
+    u"method", [unicode],
+    u"The name of the decorated method.")
+TEST_MESSAGE = MessageType(u"flocker:common:test:interface:message", [METHOD],)
+TEST_EXCEPTION = MessageType(u"flocker:common:test:interface:exception",
+                             [METHOD],)
+
+
+class IDummy(Interface):
+    """
+    Dummy interface with two test methods.
+    """
+    def return_method():
+        """
+        Return something.
+        """
+
+    def raise_method():
+        """
+        Raise something.
+        """
+
+
+@implementer(IDummy)
+class Dummy(object):
+    """
+    Dummy class that implements ``IDummy`` interface.
+    Implements two methods: one to return an object,
+    and the second to raise an ``Exception``.
+    """
+    def __init__(self, result):
+        self._result = result
+
+    def return_method(self):
+        return self._result
+
+    def raise_method(self):
+        raise self._result
+
+
+def _test_logged_method(method_name, original_name):
+    """
+    Decorator that logs a message to ``Eliot`` logger in two flows:
+    1) Log before calling given ``method_name``.
+    2) Log if ``method_name`` resulted in an ``Exception``.
```

Comment:
reStructuredText has a syntax for numbered lists - http://docutils.sourceforge.net/docs/user/rst/quickref.html#enumerated-lists - it might make sense to use it.  Also, a popular convention for docstrings is that the first sentence (ideally one line, but that's harder) is a summary of what the thing does.  Then a following paragraph offers more details.  It's kind of a nice way to arrange things but it's not mandatory.